modules/hash_demulti/bff.nf

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env nextflow
+nextflow.enable.dsl=2
+
+process bff{
+    publishDir "$projectDir/$params.outdir/$params.mode/hash_demulti/bff", mode:'copy'
+    input:
+        each seurat_object
+        //Seurat process
+        each assay
+        each methods
+        each methodsForConsensus
+        each cellbarcodeWhitelist
+        each metricsFile
+        each doTSNE
+        each doHeatmap
+        each perCellSaturation
+        each majorityConsensusThreshold
+        each chemistry
+        each callerDisagreementThreshold
+        each assignmentOutBff
+        
+    output:
+        path "bff_${task.index}"
+        
+    script:
+        def generateGenderPlot = generate_gender_plot != 'None' ? " --generateGenderPlot ${generate_gender_plot}" : ''
+        """
+        mkdir bff_${task.index}
+        bff.R --seuratObject $seurat_object --assay $assay --methods $methods --methodsForConsensus $methodsForConsensus \
```

Comment:
please dont forget to make bff.R executable otherwise we can not call the rscript without Rscript command, we will get permission denied error